sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -49,8 +49,16 @@ object DateTimeUtils {
   final val TimeZoneUTC = TimeZone.getTimeZone("UTC")
 
   val TIMEZONE_OPTION = "timeZone"
+  val TIME_ZONE_ID = "(\\+|\\-)\\d:".r
 
-  def getZoneId(timeZoneId: String): ZoneId = ZoneId.of(timeZoneId, ZoneId.SHORT_IDS)
+  def getZoneId(timeZoneId: String): ZoneId = {
```

Comment:
This change affects all the datetime related functionalities, so I think we need more tests and document updates.